mne/epochs.py

```diff
@@ -1420,6 +1420,12 @@ def next(self, return_event_id=False):
 
         return epoch if not return_event_id else epoch, self.event_id
 
+    @property
+    def times(self):
+        """Time vector in seconds."""
+        self._times.flags['WRITEABLE'] = False
```

Comment:
I wouldn't set it here, but rather in the `Epochs` constructor. Otherwise on every access it gets set again, which seems unnecessary

Reply:
the problem is that we would then need to keep track of every instance, in which `Epochs._times` is changed, which is quite a lot outside of the constructor.

1. `Epochs.decimate`: lines 564 and 567
2. `Epochs.crop`: line 1671
3. `mne.filters.resample`: line 2106
4. test_epochs: lines 2010, 2015 and 2141

so, we would need to make sure to set this flag again (which means writing unit tests for all these cases!) and keep that in mind for further modifications to the code that mess around with the times field. another possibility would be to change the whole implementation how times are represented in `Epochs`. although i think this would be the best option, this looks like a lot of work for which i do not have the time.

this is the reason why i opted for the current implementation: it is safe, tested easily and quick to implement.

Reply:
> the problem is that we would then need to keep track of every instance, in which Epochs._times is changed, which is quite a lot outside of the constructor...
> another possibility would be to change the whole implementation how times are represented in Epochs. although i think this would be the best option, this looks like a lot of work for which i do not have the time.

I would just make a `BaseEpochs._set_times(times)` method that sets the flag. It's sort of in between but it allows us to set the flag just in `_set_times` rather than in the `property`, which seems cleaner.

> (which means writing unit tests for all these cases!)

With my proposal above I don't think you need to add any more tests than what you already have done. All those other properties are implicitly tested enough.

Reply:
> but that also means that seemingly small and simple modifications to the source code result in lots of work.

Yes unfortunately long-term maintainability consideration can make what appear to be simple PRs not so simple :(

I'll push directly to your branch